crates/forge/src/lib.rs

```diff
@@ -603,6 +605,14 @@ fn run_with_fuzzing(
                 .count(),
         )?;
 
+        if let Some(result) = results
+            .iter()
+            .find(|item| matches!(item, TestCaseSummary::Failed { .. }))
+        {
+            let result = result.clone().with_runs(runs);
+            return Ok(result);
+        }
+
         if let Some(result) = results.iter().find(|item| {
             matches!(
                 item,
```

Comment:
Just realised: we shouldn't match for `Interrupted` below - if we ever return interrupted from this function it is a bug (it should return Failed instead)